### DISCUSSION TITLE
I2S stream to specific IP Address, not webserver

### DISCUSSION THREAD

**Discussion by DansDesigns** [pschatzmann/arduino-audio-tools#872] [Q&A] [answered]
Hi,
First i would like to say thank you for this absolutely amazing library!! 
I have spent about a week going through all the examples and closed issues and I'm incredibly impressed by all the work here!

I am trying to find a way to stream from an I2S Microphone (INMP441) to a specific IP Address (which is actually an MQTT Server).. 
I have seen the Stream-I2S-webserver but, if i am correct, this is to the IP Address of the ESP32? is it possible to re-direct the Stream to another, preferably external, IP Address? 

Thank you so much!

**Comment by pschatzmann** [comment, marked as answer]
There are some examples that show how to copy data using pure tcp/ip in the communications examples.
However if you want to use MQTT you need to use a MQTT library with a MQTTClient. Just learn how to use this outside of my library first. If you managed to understand how this works it should be easy to copy the data. MQTTClient inherits from Stream, so it can be used as copy destination and you could use StreamCopy to copy the data, but you don't have to....

**Reply by DansDesigns** [reply]
Hey, thanks for the reply and info! Have been away from the project for a while but getting back into it now.. I like the simple solution you present at the end, it sounds like it will accomplish what i need to do.
